sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -64,7 +64,13 @@ private String generateBinaryQuery(@NonNull Criteria criteria, @NonNull List<Pai
         parameters.add(Pair.of(parameter, subjectValue));
 
         if (CriteriaType.isFunction(criteria.getType())) {
-            return getFunctionCondition(ignoreCase, sqlKeyword, subject, parameter);
+            return getFunctionCondition(ignoreCase, sqlKeyword, subject, parameter,
+                CriteriaType.isFunctionWithCaseSensitiveSupport(criteria.getType()));
+        } else if (criteria.getType() == CriteriaType.IS_EQUAL
+                && ignoreCase != Part.IgnoreCaseType.NEVER
+                && subjectValue.getClass().getTypeName().equals("java.lang.String")) {
+            return getFunctionCondition(ignoreCase, CriteriaType.STRING_EQUAL.getSqlKeyword(),
```

Comment:
what would happen if the value is null? 

for example, if I have the following document:
```
   {
    "id": "id1",
    "mypk": "mypk1",
    "extraProperty": null,
}
```

select * from r where r.extraProperty = null will return the item
select * from r where STRINGEQUALS(r.extraProperty, null) will not return anything

Reply:
Good catch

Reply:
Fixed